tests/kernel/mem_protect/userspace/src/main.c

```diff
@@ -951,6 +959,49 @@ void test_tls_pointer(void)
 	}
 }
 
+void user_access_queue_private_data(void *p1, void *p2, void *p3)
+{
+	expect_fault = true;
+	expected_reason = K_ERR_KERNEL_OOPS;
+
+	/* try to access to private kernel data, will happen kernel oops */
+	k_queue_is_empty(&pqueue);
```

Comment:
My understanding is that kernel oops is expected behavior. But, how does sanitycheck can know if the kernel ops really happens to decide is the test case fails or not?

Reply:
Function supplied with the arguments for the function `void k_sys_fatal_error_handler`
```	
expect_fault = true;
expected_reason = K_ERR_KERNEL_OOPS;
```
If happened error has the same reason as the expected reason, then test passes.